src/privatesend.cpp

```diff
@@ -300,69 +367,18 @@ std::string CPrivateSend::GetMessageByID(PoolMessage nMessageID)
     }
 }
 
-bool CDarksendQueue::Sign()
-{
-    if(!fMasterNode) return false;
-
-    std::string strMessage = vin.ToString() + boost::lexical_cast<std::string>(nDenom) + boost::lexical_cast<std::string>(nTime) + boost::lexical_cast<std::string>(fReady);
-
-    if(!CMessageSigner::SignMessage(strMessage, vchSig, activeMasternode.keyMasternode)) {
-        LogPrintf("CDarksendQueue::Sign -- SignMessage() failed, %s\n", ToString());
-        return false;
-    }
-
-    return CheckSignature(activeMasternode.pubKeyMasternode);
-}
-
-bool CDarksendQueue::CheckSignature(const CPubKey& pubKeyMasternode)
-{
-    std::string strMessage = vin.ToString() + boost::lexical_cast<std::string>(nDenom) + boost::lexical_cast<std::string>(nTime) + boost::lexical_cast<std::string>(fReady);
-    std::string strError = "";
-
-    if(!CMessageSigner::VerifyMessage(pubKeyMasternode, vchSig, strMessage, strError)) {
-        LogPrintf("CDarksendQueue::CheckSignature -- Got bad Masternode queue signature: %s; error: %s\n", ToString(), strError);
-        return false;
-    }
-
-    return true;
-}
-
-bool CDarksendQueue::Relay()
+void CPrivateSend::AddDSTX(const CDarksendBroadcastTx& dstx)
 {
-    std::vector<CNode*> vNodesCopy = CopyNodeVector();
-    BOOST_FOREACH(CNode* pnode, vNodesCopy)
-        if(pnode->nVersion >= MIN_PRIVATESEND_PEER_PROTO_VERSION)
-            pnode->PushMessage(NetMsgType::DSQUEUE, (*this));
-
-    ReleaseNodeVector(vNodesCopy);
-    return true;
+    LOCK(cs_mapdstx);
+    mapDSTX.insert(std::make_pair(dstx.tx.GetHash(), dstx));
 }
 
```

Comment:
"Since c++11 there's no need to take a lock on map::find()" - Not sure we should rely on the specification here. We need to support some quite old compilers that clearly have some issues with C++-11 (non-working std::regex, for example).